src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -3437,6 +3438,9 @@ def update_network_plugin_settings(self, mc: ManagedCluster) -> ManagedCluster:
         if network_policy:
             mc.network_profile.network_policy = network_policy
 
+        if network_dataplane == CONST_NETWORK_DATAPLANE_CILIUM:
```

Comment:
suggest setting this only when `network_policy` isn't explicitly set.

 ```python
network_policy = self.context.get_network_policy()
if network_policy:
    mc.network_profile.network_policy = network_policy
elif network_dataplane == CONST_NETWORK_DATAPLANE_CILIUM:
    mc.network_profile.network_policy = CONST_NETWORK_POLICY_CILIUM
 ```
 
 preferrably with a comment explaining why this is necessary.

Reply:
Thank you, updated accordingly